tenzir.yaml.example

```diff
@@ -77,6 +77,10 @@ tenzir:
   # Maximum number of log messages in the logger queue.
   log-queue-size: 1000000
 
+  # Behavior if the logger queue runs full. Possible values are 'overrun_oldest'
+  # and 'block'.
+  log-overflow-policy: overrun_oldest
```

Comment:
Should this really be configurable? Why would you ever use `block`?

Reply:
Just from the behavior `block` actually makes more sense to me, since it doesn't lose any log data. In theory "block" shouldn't mean "deadlock", it should just pause the process until the logger thread has had time to catch up. I didn't investigate why exactly that didn't happen in our case.

But the real reason why I made it configurable is that we're changing the default, and this way we can easily revert to the previous behavior if the logging starts to break mysteriously afterwards.

Reply:
Then asking the question the other way around: Why would a user not want to use `block`?

Reply:
That was already answered above:

> In theory "block" shouldn't mean "deadlock", it should just pause the process until the logger thread has had time to catch up. I didn't investigate why exactly that didn't happen in our case.

Empirically `block` causes the server to lock up when the queue size limit gets hit.

Reply:
Exposing this option seems a bit like shifting the responsibility to the user. For such system internals, how would a user know what value to choose?

I think we should not expose an option to our users that we don't have tests for and know to sometimes cause hangs.

Reply:
The reason for exposing it was also already stated:

> But the real reason why I made it configurable is that we're changing the default, and this way we can easily revert to the previous behavior if the logging starts to break mysteriously afterwards.

But I agree with your reasoning as well. Wdyt of *not* documenting this knob and removing it again after ~2 releases?

Reply:
I'd rather remove it entirely, document the changed behavior, and if a user asks for it to be made configurable we bring the option back.

Reply:
If things don't go as expected we'd have to roll out and distribute new builds in an emergency. Having one option less does not justify that risk.

Reply:
We decided to keep it as a hidden option temporarily.